irc/nickname.go

```diff
@@ -39,66 +45,46 @@ func nickHandler(server *Server, client *Client, msg ircmsg.IrcMessage) bool {
 		return false
 	}
 
-	if client.nick == nickname {
+	if target.Nick() == nicknameRaw {
 		return false
 	}
 
-	// bleh, this will be replaced and done below
-	if client.registered {
-		err = client.ChangeNickname(nicknameRaw)
-	} else {
-		err = client.SetNickname(nicknameRaw)
-	}
+	hadNick := target.HasNick()
+	origNick := target.Nick()
+	origNickMask := target.NickMaskString()
+	err = client.server.clients.SetNick(target, nickname)
 	if err == ErrNicknameInUse {
 		client.Send(nil, server.name, ERR_NICKNAMEINUSE, client.nick, nicknameRaw, "Nickname is already in use")
 		return false
 	} else if err != nil {
 		client.Send(nil, server.name, ERR_UNKNOWNERROR, client.nick, "NICK", fmt.Sprintf("Could not set or change nickname: %s", err.Error()))
 		return false
 	}
-	if client.registered {
-		client.server.monitorManager.AlertAbout(client, true)
+
+	client.server.logger.Debug("nick", fmt.Sprintf("%s changed nickname to %s", origNickMask, nickname))
```

Comment:
I think it makes sense to record both the actual and casefolded nicknames the client's now usin,g to help in cases where either the casefolded or actual nick is confusing. Maybe something along the lines of this?
`"%s changed nickname to %s [%s]"` (origNickMask, nicknameRaw, nickname)

Also moves the raw nickname to the start, with the casefolded name in square brackets. This is because the raw nick is the 'preferred' one, so I think makes more sense to display as-is (same as the snomask below).